File: CN/java/subnetting/subnetting.java
import java.util.*;

class subnetting {
    public static void main(String args[])
    {
        List<Integer> hosts = new ArrayList<Integer>();
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the base class IP Address : ");
        String s = sc.next();
        System.out.println("Enter the Number of Subnets Required:");
        int a=sc.nextInt();
        for(int i=0;i<a;i++)
        {
            System.out.println("Enter Number of Hosts in Subnet-"+(i+1)+":");
            hosts.add(sc.nextInt());
        }
        int t = 0;
        for(int k : hosts)
        {
            t+=k;
        }
        if(t>252)
        {
            System.out.println("Total Host must be less than or equal to 252");
        }
        else
        {
            int prev = 0;
            for(int k = 0; k<hosts.size(); k++)
            {
                int minHost = (int) Math.pow(2, Math.ceil(Math.log(hosts.get(k)+2) / Math.log(2)));
                System.out.println("Range for Subnet-"+k+" is "+s+"."+prev+"-"+(prev+minHost-1));
                prev+=minHost;
            }
        }
    }
}

File: CN/python/subnetting/subnetting.py
import math

l=[]
s=input("Enter the base class ip address : ")
n=int(input("Enter the no. of usbnets required : "))
for i  in range(n):
    mess = "Enter the number of hosts in subnet - "+str(i+1)+" : "
    l.append(int(input(mess)))
total = 0
for i in l:
    total+=i

if(total>252):
    print("Total hosts must be less than or equal to 252")
else:
    prev=0
    for i in range(n):
        minHost = 2**math.ceil(math.log(l[i]+2)/math.log(2))
        print("Range for subnet - ",i+1," is ",s,".",prev,"-",(prev+minHost-1))
        prev+=minHost





